tests/app/main/test_frameworks.py

```diff
@@ -1778,29 +1937,117 @@ def test_dos_drafts_list_with_closed_framework(self, count_unanswered, data_api_
         supplier_framework['declaration']['nameOfOrganisation'] = u'£unicodename'
         return_supplier_framework.return_value = supplier_framework
 
-        res = self.client.get("/suppliers/frameworks/g-cloud-8/signer-details")
+        res = self.client.get("/suppliers/frameworks/g-cloud-8/234/signer-details")
         page = res.get_data(as_text=True)
         assert res.status_code == 200
         assert u'Details of the person who is signing on behalf of £unicodename' in page
 
+    def test_signer_details_shows_existing_signer_details(self, return_supplier_framework, data_api_client):
+        self.login()
+        data_api_client.get_framework.return_value = get_g_cloud_8()
+        data_api_client.get_framework_agreement.return_value = self.framework_agreement(
+            signed_agreement_details={
+                "signerName": "Sid James",
+                "signerRole": "Ex funny man"
+            }
+        )
+        supplier_framework = self.supplier_framework(framework_slug='g-cloud-8', on_framework=True)['frameworkInterest']
+        return_supplier_framework.return_value = supplier_framework
+
+        res = self.client.get("/suppliers/frameworks/g-cloud-8/234/signer-details")
+        page = res.get_data(as_text=True)
+        assert res.status_code == 200
+        assert "Sid James" in page
+        assert "Ex funny man" in page
+
+    def test_404_if_framework_in_wrong_state(self, return_supplier_framework, data_api_client):
+        self.login()
+        # Suppliers can only sign agreements in 'standstill' and 'live' lifecycle statuses
+        data_api_client.get_framework.return_value = self.framework(status='pending')
+        data_api_client.get_framework_agreement.return_value = self.framework_agreement()
+        supplier_framework = self.supplier_framework(framework_slug='g-cloud-8', on_framework=True)['frameworkInterest']
+        return_supplier_framework.return_value = supplier_framework
+
+        res = self.client.get("/suppliers/frameworks/g-cloud-8/234/signer-details")
+        assert res.status_code == 404
+
+    @mock.patch('app.main.views.frameworks.check_agreement_is_related_to_supplier_framework_or_abort')
+    def test_we_abort_if_agreement_does_not_match_supplier_framework(
+        self, check_agreement_is_related_to_supplier_framework_or_abort, return_supplier_framework, data_api_client
+    ):
+        self.login()
+        data_api_client.get_framework.return_value = get_g_cloud_8()
+        data_api_client.get_framework_agreement.return_value = self.framework_agreement(supplier_id=2345)
+        supplier_framework = self.supplier_framework(framework_slug='g-cloud-8', on_framework=True)['frameworkInterest']
+        return_supplier_framework.return_value = supplier_framework
+
+        res = self.client.get("/suppliers/frameworks/g-cloud-8/234/signer-details")
+        check_agreement_is_related_to_supplier_framework_or_abort.assert_called_with(
```

Comment:
Does this actually test that it's aborted? It looks like it just checks if the method was called.